h5py/_hl/selections.py

```diff
@@ -17,10 +17,9 @@
 import numpy as np
 
 from .base import product
-from .. import h5s, h5r
+from .. import h5s, h5r, _selector
 
-
-def select(shape, args, dsid):
+def select(shape, args, dataset=None):
```

Comment:
Even though this is technically private API we should our users are clever and found it.  I think we should document the (internal) API changes out of an abundance of caution.